src/lib/receiver.js

```diff
@@ -185,6 +193,18 @@ function createReceiver (opts) {
       return rejectIncomingTransfer(transfer.id, 'no-packet')
     }
 
+    // check if the address ends with "~ipr" + our receiver id +
+    const myRequestAddressRegex = new RegExp(receiverId + '\\..+$')
```

Comment:
Good catch! That would matter. Because it requires the next character after the wildcard character to be a period `.` - which it won't be if the period was already matched by the wildcard. So yeah, I don't think this code works, unless I'm missing something.